docker/entrypoint.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+
+# run aesmd in the background
+/opt/intel/sgx-aesm-service/aesm/aesm_service
```

Comment:
This will only work on sgx hardware, correct?

Reply:
I do not think so, as the `SGX_SDK` libs should provide the required `*_sim` libraries, but let me get back to you in this after I tested it. :+1:

Reply:
Well, some not so great news:
- In `SW` mode the `enclave.qe_get_target_info().unwrap();` line causes a panic:
```rust
[load_pce ../pce_wrapper.cpp:195] Error, call sgx_create_enclave for PCE fail [load_pce], SGXError:200e.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: SgxQuote(SGX_QL_INTERFACE_UNAVAILABLE)', service/src/main.rs:177:68
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I think that refactoring it in a way that:
- This becomes an `Option<T>` for the call site
- This is only used if `--skip-ra` is not present

Should resolve this issue, let me give it a try.

Reply:
Well, this was a rather complicated journey, but long story short DCAP with SW mode works now, thanks for the check!